### PR TITLE
GH-4: map marker icons crash

### PR DESCRIPTION
**Pull request by MarcelReiter**
Attempts to adress #4.
The main issue was:
The BitmapFactory required the map to be initialised already.

### CODE REVIEW

File: app/android/app/src/main/java/com/github/tscholze/duobahn/data/domain/models/MarkerDefinition.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.github.tscholze.duobahn.data.domain.models
+
+import com.google.android.libraries.maps.model.LatLng
+
+// TODO is this really a domain model?
```
Comment (tscholze) [inline app/android/app/src/main/java/com/github/tscholze/duobahn/data/domain/models/MarkerDefinition.kt:5]: What else, its not a DTO. *hm*

File: app/android/app/src/main/java/com/github/tscholze/duobahn/data/domain/models/MarkerDefinition.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+data class MarkerDefinition(
+    val type: MarkerType,
+    val title: String,
+    val coordinate: LatLng,
+    val snippet: String?,
+) {
+    enum class MarkerType { WEBCAM, ROADWORK }
```
Comment (tscholze) [inline app/android/app/src/main/java/com/github/tscholze/duobahn/data/domain/models/MarkerDefinition.kt:13]: Okay this is some Kotlin magic I have to look into it.

I thought that such things were sealed classes aka "rich enums"?

**Review by tscholze — APPROVED** [review]

File: app/android/app/src/main/java/com/github/tscholze/duobahn/data/domain/models/MarkerDefinition.kt (hunk 2 above)
Comment (MarcelReiter) [reply app/android/app/src/main/java/com/github/tscholze/duobahn/data/domain/models/MarkerDefinition.kt:13]: To be honest, not much magic here:

[Data classes](https://kotlinlang.org/docs/data-classes.html) are cool, but you already seem know them, e.g. from the Autobahn data class:
Main benefits are: Kotlin auto-generates `equals()` & `hashCode()` functions for them, and they can be used for [destructuring](https://kotlinlang.org/docs/destructuring-declarations.html).

And [enum classes](https://kotlinlang.org/docs/enum-classes.html) are just Kotlins syntax for enum.
Main benefit is that they force handling of each single entry:
This avoids a szenario, where you add a MarkerType GAS_STATION, but forget to provide an image in `MapView`.

File: app/android/app/src/main/java/com/github/tscholze/duobahn/data/domain/models/MarkerDefinition.kt (hunk 1 above)
Comment (MarcelReiter) [reply app/android/app/src/main/java/com/github/tscholze/duobahn/data/domain/models/MarkerDefinition.kt:5]: Well, somehow it is a DTO, but between `viewModel `and `View `😓

**Pull request merged**